sbt-scala-native/src/main/scala/scala/scalanative/sbtplugin/ScalaNativePluginInternal.scala

```diff
@@ -61,11 +70,38 @@ object ScalaNativePluginInternal {
     }
   }
 
+  case object NoPosition extends Position {
+    import Maybe._
+
+    def line: Maybe[Integer]        = nothing[Integer]
+    def lineContent: String         = ""
+    def offset: Maybe[Integer]      = nothing[Integer]
+    def pointer: Maybe[Integer]     = nothing[Integer]
+    def pointerSpace: Maybe[String] = nothing[String]
+    def sourcePath: Maybe[String]   = nothing[String]
+    def sourceFile: Maybe[File]     = nothing[File]
+  }
+
   /** Compiles application nir to llvm ir. */
-  private def compileNir(config: tools.Config): Seq[nir.Attr.Link] = {
-    val driver       = tools.Driver(config)
-    val (links, raw) = tools.link(config, driver)
-    val optimized    = tools.optimize(config, driver, raw)
+  private def compileNir(config: tools.Config, logger: Logger, reporter: Option[Reporter]): Seq[nir.Attr.Link] = {
+    val driver = tools.Driver(config)
+    val (unresolved, links, raw) = tools.link(config, driver)
+
+    if(!unresolved.isEmpty) {
+      import nir.Shows._
+
+      unresolved.map(u => sh"$u".toString).sorted.foreach{signature =>
+        val message = s"cannot link: $signature"
+        reporter match {
+          case Some(r) => r.log(NoPosition, message, Severity.Error)
+          case None => logger.error(message)
+        }        
+      }
```

Comment:
Now it's a bit too much, but if we add position to linker error it will actually provide lots os things for free.